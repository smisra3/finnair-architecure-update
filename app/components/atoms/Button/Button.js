// @flow
import type { Node } from 'react';

import styled from 'styled-components';
import styles from './Button.style';
import Spinner from '../../atoms/Spinner';

type Props = {
  className: string,
  children: Node,
  inheritedStyles?: string,
  type?: string,
  primary?: boolean,
  secondary?: boolean,
  elementRef?: any => void,
  loader?: boolean,
  loaderPositionCenter?: boolean,
  disabled?: boolean,
  ariaLabel?: string,
};

const Button = ({
  className,
  children,
  inheritedStyles,
  type,
  primary,
  secondary,
  elementRef,
  loader,
  loaderPositionCenter,
  disabled,
  ariaLabel,
  ...others
}: Props): Node => (
  <button
    className={className}
    type={type}
    {...others}
    ref={elementRef}
    disabled={disabled || loader}
    aria-label={ariaLabel || null}
  >
    {loader && (
    <div className={`spinner ${loaderPositionCenter ? 'center-spinner' : ''}`}>
      <Spinner overlayEnabled spinner />
    </div>
      )}
    {children}
  </button>
);

Button.defaultProps = {
  inheritedStyles: '',
  type: 'button',
  primary: false,
  secondary: false,
  elementRef: undefined,
  loader: false,
  disabled: false,
  loaderPositionCenter: false,
  ariaLabel: '',
};

export default styled(Button)`${styles}`;
export { Button as ButtonVanilla };
