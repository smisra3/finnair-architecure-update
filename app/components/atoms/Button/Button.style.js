import { css } from 'styled-components';

const ButtonStyles = css`
  border: none;
  border-radius: 0;
  background: none;
  position: relative;
  min-height: 48px;
  cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
  opacity: ${props => (props.disabled ? '0.3' : '1')};
  ${props =>
    (props.primary
      ? `
        background: ${props.theme.buttonBackgroundColorPrimary} ;
        color: ${props.theme.buttonColorPrimary};
        font-size: ${props.theme.fontSizeButtonPrimary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.secondary
      ? `
        background: ${props.theme.buttonBackgroundColorSecondary} ;
        color: ${props.theme.buttonColorSecondary};
        font-size: ${props.theme.fontSizeButtonSecondary};
        padding: 0 1rem;
      ` : '')};
  ${props =>
    (props.fullWidth
      ? `
        width: 100% ;
      ` : '')};

  &:focus {
    outline: 1px dashed ${props => props.theme.colorOutline};
  }

  &:focus:not(.focus-visible) {
    outline: none;
  }

  .spinner {
    position: absolute;
    top: 50%;
    left: 9px;
    transform: translateY(-50%);

    &.center-spinner {
      left: 50%;
      transform: translateX(-50%) translateY(-50%);
    }

    div {
      height: 25px;
      width: 25px;
      top: 0;
    }
  }
`;

export default ButtonStyles;
