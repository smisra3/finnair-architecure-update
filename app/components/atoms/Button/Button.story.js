// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';
import {
  primaryButton,
  primaryButtonDisabled,
  secondaryButton,
  secondaryButtonDisabled,
} from './Button.mock';

// Import Vanilla Component to display markup and props
import { ButtonVanilla } from './Button';
// Import Styled Component to showcase variations
import Button from './index';

storiesOf('Atoms', module).addWithChapters('Buttons', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ButtonVanilla {...primaryButton} className={`hide-default-sample ${primaryButton.className}`}>Submit</ButtonVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Primary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...primaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...primaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
    {
      title: 'Secondary Button Variations',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Button {...secondaryButton}>Submit</Button>,
        },
        {
          title: 'Disabled',
          sectionFn: () => <Button {...secondaryButtonDisabled}>Submit</Button>,
        },
      ],
    },
  ],
});
