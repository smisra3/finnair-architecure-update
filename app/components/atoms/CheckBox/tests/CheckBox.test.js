import { shallow } from 'enzyme';
import 'jest-styled-components';
import CheckBoxStyle, { CheckBoxVanilla as CheckBox } from '../CheckBox';
import { ButtonVanilla } from '../../Button/Button';


describe('CheckBox Component with style ', () => {
  test('should render correctly with styles', () => {
    let checkBox = '';
    checkBox = shallow(<CheckBoxStyle />);
    expect(checkBox).toMatchSnapshot();
  });
});

describe('CheckBox Component ', () => {
  test('should render correctly', () => {
    let checkBox = '';
    checkBox = shallow(<CheckBox />);
    expect(checkBox).toMatchSnapshot();
  });
  test('should render checkbox checked', () => {
    const checkBoxChecked = shallow(<CheckBox isChecked />);
    expect(checkBoxChecked).toMatchSnapshot();
  });
  test('should render checkbox without label', () => {
    const checkBoxChecked = shallow(<CheckBox hideLabel />);
    expect(checkBoxChecked).toMatchSnapshot();
  });
  test('should trigger focus event', () => {
    const checkBoxChecked = shallow(<CheckBox hideLabel />);
    checkBoxChecked.find('input').simulate('focus');
    expect(checkBoxChecked.state().isFocused).toEqual(true);
  });

  test('should trigger blur State', () => {
    const checkBoxChecked = shallow(<CheckBox hideLabel />);
    checkBoxChecked.find('input').simulate('blur');
    expect(checkBoxChecked.state().isFocused).toEqual(false);
  });

  test('should render string label when createLabelField is invoked with hideLabel="true"', () => {
    const checkBoxChecked = shallow(<CheckBox hideLabel label="abc" />);
    expect(checkBoxChecked.containsMatchingElement(<span className="label-field is-hidden">abc</span>)).toBe(true);
  });

  test('should render string field label when createLabelField is invoked hideLabel="false"', () => {
    const checkBoxChecked = shallow(<CheckBox label={ButtonVanilla} />);
    expect(checkBoxChecked.find(ButtonVanilla)).toHaveLength(1);
  });
});
