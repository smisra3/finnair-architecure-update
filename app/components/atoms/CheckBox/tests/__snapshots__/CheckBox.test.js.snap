// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Component  should render checkbox checked 1`] = `
<div>
  <input
    aria-label=""
    checked={true}
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <label
    className=" is-checked"
  >
    <span
      aria-hidden={false}
      className="checkbox checkbox-tick checkbox-checked"
      data-slnm-id="checkbox"
    />
  </label>
</div>
`;

exports[`CheckBox Component  should render checkbox without label 1`] = `
<div>
  <input
    aria-label=""
    checked={false}
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <label
    className=""
  >
    <span
      aria-hidden={false}
      className="checkbox checkbox-tick  checkbox-unchecked"
      data-slnm-id="checkbox"
    />
  </label>
</div>
`;

exports[`CheckBox Component  should render correctly 1`] = `
<div>
  <input
    aria-label=""
    checked={false}
    disabled={false}
    onBlur={[Function]}
    onFocus={[Function]}
    readOnly={false}
    type="checkbox"
  />
  <label
    className=""
  >
    <span
      aria-hidden={false}
      className="checkbox checkbox-tick  checkbox-unchecked"
      data-slnm-id="checkbox"
    />
  </label>
</div>
`;

exports[`CheckBox Component with style  should render correctly with styles 1`] = `
.c0 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.c0 label::before {
  position: absolute;
  left: 0;
  top: calc(50% - 12px);
  content: "";
  height: 20px;
  width: 20px;
  border: 1px solid;
  border-radius: 1px;
  background: #fff;
  box-sizing: border-box;
}

.c0 label::after {
  position: absolute;
  left: 6px;
  top: calc(50% - 7px);
  content: "";
  display: inline-block;
  height: 5px;
  width: 10px;
  border-left: 1px solid;
  border-bottom: 1px solid;
  -webkit-transform: rotate(-45deg);
  -ms-transform: rotate(-45deg);
  transform: rotate(-45deg);
  box-sizing: border-box;
}

.c0 .label-field {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  word-break: break-word;
  margin-left: 10px;
  position: relative;
  top: -4px;
}

.c0 ~ input {
  position: absolute;
  width: 15px;
  height: 15px;
  opacity: 0;
  z-index: -1;
}

.c0 input + label::after {
  content: none;
}

.c0 input:checked + label::after {
  content: "";
}

.c0 input:focus + label::before {
  border: 2px solid;
  outline: 1px solid;
}

<CheckBox
  checkedClass=""
  className="c0"
  dataSlnmId="checkbox"
  disabled={false}
  imageUrl=""
  inheritedStyles=""
  inputAriaLabel=""
  isChecked={false}
  isCustom={false}
  label={null}
  labelOnly={false}
  onBlur={[Function]}
  onFocus={[Function]}
  readOnly={false}
  uncheckedClass=""
/>
`;
